### ISSUE TITLE
tf.py_function InternalError in distributed mode

### ISSUE THREAD

**Issue by em1208** [tensorflow/tensorflow#27102]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
Linux Ubuntu 16.04.6 LTS
- TensorFlow installed from (source or binary):
pip
- TensorFlow version (use command below):
1.13.1
- Python version:
3.6.7

**Describe the current behavior**
I'm testing tensorflow distributed on the same machine. I opened two separate shells and created a cluster using:

```python
import tensorflow as tf  
cluster = tf.train.ClusterSpec({"local": ["localhost:2222", "localhost:2223"]})
server = tf.train.Server(cluster, job_name="local", task_index=0)
```

on shell 0 and

```python
import tensorflow as tf
cluster = tf.train.ClusterSpec({"local": ["localhost:2222", "localhost:2223"]})
server = tf.train.Server(cluster, job_name="local", task_index=1)
```

on shell 1.

On shell 1 I ran:

```python
import tensorflow as tf
import numpy as np

def add(a, b):
    return a + b

NUM = 1000
ones = np.ones((NUM))

graph = tf.Graph()
with graph.as_default():
    va = tf.Variable(ones)
    vb = tf.Variable(ones)
    with tf.device("job:local/task:1"):
        inputs = [va, vb]
        out = tf.py_function(add, inputs, tf.float64)

    with tf.Session("grpc://localhost:2223", graph=graph) as sess:
        sess.run(tf.global_variables_initializer())
        print(sess.run([va, vb]))
        print(sess.run([out]))
```

which returns the error:

> InternalError (see above for traceback): expected the py_func to return a Tensor backed by memory in /job:local/replica:0/task:1/device:CPU:0, but is actually in /job:localhost/replica:0/task:0/device:CPU:0. This is a bug.

while this does not happen if I replace tf.py_function in the code above with tf.py_func.

**Describe the expected behavior**

Code should run successfully like when using tf.py_func and should print the arrays:

> [array([1., 1., 1., 1., ..., 1., 1., 1., 1.]), array([1., 1., 1., 1., ..., 1., 1., 1., 1.])]
> [array([2., 2., 2., 2., ..., 2., 2., 2., 2.])]

**Comment by alextp** [comment]
I believe this has been fixed in nightly.

Can you try?

**Comment by em1208** [comment]
Hi @alextp,

I confirm this issue has been fixed on 1.14.1-dev20190325. Thanks!
